File: src/main/java/com/gmail/borlandlp/minigamesdtools/arena/gui/providers/GUIProvider.java
package com.gmail.borlandlp.minigamesdtools.arena.gui.providers;

import com.gmail.borlandlp.minigamesdtools.arena.ArenaBase;
import com.gmail.borlandlp.minigamesdtools.arena.ArenaPhaseComponent;

public abstract class GUIProvider implements ArenaPhaseComponent {
    private ArenaBase arena;

    public GUIProvider(ArenaBase arenaBase) {
        this.arena = arenaBase;
    }

    public GUIProvider() {}

    public ArenaBase getArena() {
        return arena;
    }

    public void setArena(ArenaBase arena) {
        this.arena = arena;
    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/arena/gui/providers/GUIProvider.kt
package com.gmail.borlandlp.minigamesdtools.arena.gui.providers

import com.gmail.borlandlp.minigamesdtools.arena.ArenaBase
import com.gmail.borlandlp.minigamesdtools.arena.ArenaPhaseComponent

abstract class GUIProvider : ArenaPhaseComponent {
    var arena: ArenaBase? = null

    constructor(arenaBase: ArenaBase?) {
        arena = arenaBase
    }

    constructor() {}

}